Src/FluentAssertions/Equivalency/EnumerableEquivalencyValidator.cs

```diff
@@ -54,7 +54,7 @@ public void Execute<T>(object[] subject, T[] expectation)
 
         private bool AssertIsNotNull(object expectation, object[] subject)
         {
-            return AssertionScope.Current
+            return expectation != null || AssertionScope.Current
```

Comment:
What measurable difference does the added conditional make?

Reply:
About 0.07% overall, 100% of this method only. I'll revert this.